server/pulp/server/controllers/repository.py

```diff
@@ -1272,9 +1272,10 @@ def _do_publish(repo_obj, dist_id, dist_inst, transfer_repo, conduit, call_confi
     dist = model.Distributor.objects.get_or_404(
         repo_id=repo_obj.repo_id, distributor_id=dist_id)
 
-    # Use raw pymongo not to fire the signal hander
-    model.Distributor.objects(repo_id=repo_obj.repo_id, distributor_id=dist_id).\
-        update(set__last_publish=publish_end_timestamp)
+    if publish_report.success_flag and not publish_report.canceled_flag:
```

Comment:
You don't need to check the success_flag here. If it's false, an exception is raised on line 1247.